### PR TITLE
Improve parsing slice

### PR DESCRIPTION
**Pull request by scxiao**

### CODE REVIEW

File: src/include/migraphx/op/slice.hpp [hunk 1]
```diff
@@ -11,6 +11,8 @@
 #include <migraphx/config.hpp>
 #include <cmath>
 #include <utility>
+#include <iostream>
```
Comment (kahmed10) [inline src/include/migraphx/op/slice.hpp]: I don't think this include is needed anymore.

File: src/onnx/onnx.cpp [hunk 2]
```diff
@@ -745,26 +745,72 @@ struct onnx_parser
...
         std::vector<size_t> dims = args[0]->get_shape().lens();
         size_t num_dims          = dims.size();
-        if(contains(attributes, "axes"))
+
+        // slice can have up to 5 inputs, we first check the 5th one
+        // to decide whether MIGRAPHX can handle this slice
+        if(args.size() == 5)
+        {
+            auto tind_ins = args.back();
+            if(!tind_ins->can_eval())
```
Comment (kahmed10) [inline src/onnx/onnx.cpp]: Can we make a function that checks can_eval and replace the conditionals on this line and 770 and 786?

File: src/include/migraphx/op/slice.hpp (hunk 1 above)
Comment (scxiao) [reply src/include/migraphx/op/slice.hpp]: Fixed

File: src/onnx/onnx.cpp (hunk 2 above)
Comment (scxiao) [reply src/onnx/onnx.cpp]: we have a function check_arg_empty() that can be used here.

**Pull request merged**